searx/static/themes/simple/.gitignore

```diff
@@ -1,2 +1 @@
 /node_modules
-ion.less
```

Comment:
See my comment about `searx/static/themes/oscar/generated/.gitignore`.  I know, we removed this less file in 7137d2893 but now I want to say, that this less file is _source code_ to me. May be I have a wrong view on this aspect / do you have an argument that could convince me?

Reply:
`ion.less` is the result `grunt run build`, so each `Static build` commit may include `ion.less` (not if the file is not modified).

So `src/less` has a mix of source code and generated file.

Reply:
> `ion.less` is the result `grunt run build`, if this file is committed, each `Static build` commit will include `ion.less`.

Yes, but this file won't change / by what (line) is this file created, I had a look the gruntfile but I can't find a passage where _generated_ folder is used (in "webfont).

> So `src/less` will have a mix of source code and generated files.

See my doc-example in https://github.com/searxng/searxng/pull/157#discussion_r654796752 ..  the files in docs/build-templates will also _committed each 'Doc build'_  but I decided to add them to the repository since they are needed to understand the reST document.

Reply:
What if the generated file:
* are commited in `src/generated` ?
* commited with the `Static build` commit ?

Reply:
Sorry that I hang at this point but before answering your question I will understand it all and I have two more questions:

* https://github.com/searxng/searxng/blob/eb6832eb95f064a9450f5658f3cee52d374c6a31/searx/static/themes/simple/gruntfile.js#L119-L120

If `destLess` is `/searx/static/themes/simple` whhy is the less file landing in `searx/static/themes/simple/generated`?

----

Do you know  the difference of [grunt-webfont](https://www.npmjs.com/package/grunt-webfont) and [grunt-webfonts](https://www.npmjs.com/package/grunt-webfonts) .. if I am not wrong https://github.com/sapegin/grunt-webfont is _archieved_ (last release is  years old) and https://github.com/L2jLiga/grunt-webfonts is moore up-to-date / but I also see a critical issue https://github.com/L2jLiga/grunt-webfonts/issues/20

Reply:
> What if the generated files:
> * are commited in `src/generated` ?
> * commited with the `Static build` commit ?

`src/generated`  is a good place but I wont recommend it in the `[static] make themes` commit.

1. the aim is to split the static commit from the src commit
2. these (less) files will only change when I change something in src or when I change the build rules (or I am wrong?)

Beside your _commit question_, my grunt-webfont(s) question resists, If you could have a look at it would be great / thanks!

Reply:
grunt-webfont is (one of) the last peace that need an update.
There is also https://github.com/Frontify/grunt-webfont

Reply:
> grunt-webfont is (one of) the last peace that need an update.

OK, I will wait about your next commits  .. ATM I am a bit confused: 

1. https://github.com/Frontify/grunt-webfont is a fork of
2. https://github.com/jleider/grunt-webfont which is a fork of
3. https://github.com/sapegin/grunt-webfont

Only the last one is available from npm https://www.npmjs.com/package/grunt-webfont 

.. and then there is another package with a trailing 's' at the end ..

https://www.npmjs.com/package/grunt-webfonts from https://github.com/L2jLiga/grunt-webfonts

Reply:
Another problem I have (on ubu20044):

```
> ttf2woff2@2.0.3 install /x/share/searx/searx/static/themes/simple/node_modules/ttf2woff2
> (node-gyp rebuild > builderror.log) || (exit 0)

Traceback (most recent call last):
  File "/usr/share/nodejs/node-gyp/gyp/gyp_main.py", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3254, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3237, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3266, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'gyp==0.1' distribution was not found and is required by the application
```

To decouple from OS it might be a solution  to run in pyenv / in a first try it worked for me)

```diff
diff --git a/manage b/manage
index 61376514..cbf81779 100755
--- a/manage
+++ b/manage
@@ -470,13 +470,13 @@ themes.all() {
 
 themes.oscar() {
     build_msg GRUNT "theme: oscar"
-    npm --prefix searx/static/themes/oscar run build
+    pyenv.cmd npm --prefix searx/static/themes/oscar run build
     dump_return $?
 }
 
 themes.simple() {
     build_msg GRUNT "theme: simple"
-    npm --prefix searx/static/themes/simple run build
+    pyenv.cmd npm --prefix searx/static/themes/simple run build
     dump_return $?
 }
```

But I am not very happy with this solution .. to much tools involved here: start a python process to start a grunt tporocess to start  a node implementation, which starts a python application which starts binaries tools from the OS ... pew.

Reply:
> grunt-webfont is (one of) the last peace that need an update.

Do we really need to automate the build of the font and css @font-face?  Do you expect any changes in the "ion" font?  If not, I think it is better to create the font once and add the font once to the repository in the "static/" folder.

Reply:
webfont provides [two engines](https://github.com/Frontify/grunt-webfont#available-engines):
* fontforge
* node

I've tried the node engine: the built font doesn't include magnet.svg, I've tried to modified the .svg file in different ways without fixing the issue. So it is no go.

The commit cfe850aade7b2d7b050c6683a080b77cc25d1fed includes the build fonts.

`npm run webfont` can build the fonts.

Reply:
> `npm run webfont` can build the fonts.

Yes, but with the errors I posted in https://github.com/searxng/searxng/pull/157#discussion_r654836723 (at leats on my desktops)

IMO all the (grunt, node) webfont solutions have a lack of care .. and the mix of used technologies is scary to me.

Except the Patch in my  https://github.com/searxng/searxng/pull/157#discussion_r654836723 I do not have a better solution at hand right now.  So I would suggest to leave it as-is for the first.  On the long term I guess I would have look for a solution implemented in a small  python or shell script. 

grunt-webfont implements not much more than a call of the  https://fontforge.org/docs/scripting/python.html API .. unfortunately I have no time for this at the moment :-o

Reply:
>> Do we really need to automate the build of the font and css @font-face?

> `npm run webfont` can build the fonts.

I think your solution in dca3bcca9 is the best compromise .. thanks!!